### ISSUE TITLE
Configuration for rule "<some rule>" is invalid

### ISSUE THREAD

**Issue by icecream17** [EvgenyOrekhov/eslint-config-hardcore#366]
I'm probably missing something silly (basically 0 results from google) but I got this error when using eslint-config-hardcore in a create-react-app project:
![image](https://user-images.githubusercontent.com/58114641/138195466-01eb0e77-ff85-4191-a5d6-4da871cc4a3a.png)

I also tried just using a globally installed eslint:
![image](https://user-images.githubusercontent.com/58114641/138195293-a1adcbbc-a9ec-4d11-a4bd-2baf8e30ec2b.png)

<details>
<summary>.eslintrc</summary>

```jsonc
{
   // The error also happens when I comment out every "extends" other than hardcore
  "extends": [
    "eslint:recommended",
    "react-app",
    "react-app/jest",
    "plugin:flowtype/recommended",
    // "plugin:import/recommended",
    "plugin:jsx-a11y/strict",
    "plugin:jest-dom/recommended",
    "plugin:testing-library/react",
    "plugin:react/recommended",
    "plugin:react-hooks/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "hardcore",
    "hardcore/react",
    "hardcore/react-testing-library",
    "hardcore/jest",
    "hardcore/fp",
    "hardcore/ts"
  ],
  "env": {
    "browser": true
  },
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "plugins": [
    "flowtype",
    // "import",
    "jsx-a11y",
    "jest-dom",
    "testing-library",
    "react",
    "react-hooks",
    "@typescript-eslint"
  ],
  "reportUnusedDisableDirectives": true,
  "rules": {
    "no-alert": "error", // It's deprecated
    "prefer-arrow-callback": "warn",
    "prefer-const": "warn",
    "prefer-numeric-literals": "warn",
    "prefer-promise-reject-errors": "warn",
    "prefer-regex-literals": "warn",
    "prefer-rest-params": "warn",
    "prefer-spread": "warn",


    "jest/no-commented-out-tests": "warn",
    "jest/no-identical-title": "warn",
    "jest/prefer-called-with": "warn",
    "jest/prefer-expect-assertions": "off",
    "jest/prefer-hooks-on-top": "off",
    "jest/prefer-spy-on": "warn",
    "jest/prefer-strict-equal": "warn",
    "jest/prefer-to-be-null": "error",
    "jest/prefer-to-be-undefined": "error",
    "jest/prefer-to-contain": "error",
    "jest/prefer-to-have-length": "error",
    "jest/prefer-todo": "warn",
    "jest/valid-describe": "error",
    "jest/valid-expect": "error",
    "jest/valid-expect-in-promise": "error",
    "jest/valid-title": "error",


    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/no-explicit-any": "off", // Check every once in a while
    "@typescript-eslint/unbound-method": "off", // Invalid errors - methods ARE bound silly

    "@typescript-eslint/no-invalid-this": "warn",
    "@typescript-eslint/no-invalid-void-type": "warn",
    "@typescript-eslint/prefer-enum-initializers": "warn",
    "@typescript-eslint/prefer-for-of": "warn",
    "@typescript-eslint/prefer-function-type": "warn",
    "@typescript-eslint/prefer-includes": "error",
    "@typescript-eslint/prefer-literal-enum-member": "warn",
    "@typescript-eslint/prefer-nullish-coalescing": "error",
    "@typescript-eslint/prefer-optional-chain": "error",
    "@typescript-eslint/prefer-readonly": "warn",
    "@typescript-eslint/prefer-readonly-parameter-types": "off",
    "@typescript-eslint/prefer-reduce-type-parameter": "warn",
    "@typescript-eslint/prefer-string-starts-ends-with": "warn",
    "@typescript-eslint/prefer-ts-expect-error": "warn",


    "react/button-has-type": "warn",
    "react/prefer-read-only-props": "warn",


    "testing-library/await-fire-event": "warn",
    "testing-library/prefer-explicit-assert": "warn",


    // disabling base rules
    "no-use-before-define": "off",
    "@typescript-eslint/no-use-before-define": "error"
  },
  "settings": {
    "linkComponents": [
      "ExternalLink"
    ]
  }
}
```

</details>

**Comment by EvgenyOrekhov** [comment]
That might happen if you have more than 1 version of eslint-plugin-sort-class-members in your dependencies. Try searching for `eslint-plugin-sort-class-members` in your `package-lock.json`, you will probably see more than 1 instance, one of which will probably be outdated. And that outdated version ends up having higher priority when running` eslint .`

You can try installing `eslint-plugin-sort-class-members` manually: `npm install --save-dev eslint-plugin-sort-class-members@latest`. If that alone doesn't help, then you can also try to reinstall all dependencies: `rm -rf node_modules package-lock.json && npm install`.

Let me know what you find!

**Comment by icecream17** [comment]
I tried to dedupe my dependencies, and it fixed the `eslint-plugin-sort-class-members` one, but I can't dedupe `import` because there's both "2.23.4" and "^2.24.2" required.

```lock
"@shopify/eslint-plugin@npm:^40.4.0":
  version: 40.4.0
  resolution: "@shopify/eslint-plugin@npm:40.4.0"
  dependencies:
    "@typescript-eslint/eslint-plugin": ^4.28.2
    "@typescript-eslint/parser": ^4.28.2
    babel-eslint: 10.1.0
    change-case: ^4.1.2
    common-tags: ^1.8.0
    eslint-config-prettier: 8.3.0
    eslint-module-utils: 2.6.1
    eslint-plugin-babel: 5.3.1
    eslint-plugin-eslint-comments: 3.2.0
    eslint-plugin-graphql: 4.0.0
    eslint-plugin-import: 2.23.4

"eslint-config-hardcore@npm:^22.2.0":
  version: 22.2.0
  resolution: "eslint-config-hardcore@npm:22.2.0"
  dependencies:
    "@html-eslint/eslint-plugin": ^0.11.0
    "@html-eslint/parser": ^0.11.0
    "@putout/plugin-apply-shorthand-properties": ^3.0.2
    "@shopify/eslint-plugin": ^40.4.0
    
    (later)
    
    eslint-plugin-import: ^2.24.2
```

**Comment by EvgenyOrekhov** [comment]
@icecream17 Shopify fixed their semver ranges and released a new version of `@shopify/eslint-plugin`. I just updated all dependencies and released **eslint-config-hardcore** [23.3.0](https://github.com/EvgenyOrekhov/eslint-config-hardcore/releases/tag/v23.3.0). You can try and see if it works for you now.
